### PR TITLE
feat(commerce) add `atomic-commerce-search-box-instant-products`

### PR DESCRIPTION
**Pull request by y-lakhdar**
[Add atomic-commerce-search-box-instant-products](https://coveord.atlassian.net/browse/KIT-3132)

image

### CODE REVIEW

File: packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-instant-products/atomic-commerce-search-box-instant-products.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    };
+  }
+
+  private onSuggestedQueryChange() {
+    if (
+      !this.bindings.getSuggestionElements().length &&
+      !this.bindings.searchBoxController.state.value
+    ) {
+      console.warn(
+        "There doesn't seem to be any query suggestions configured. Make sure to include either an atomic-search-box-query-suggestions or atomic-search-box-recent-queries in your search box in order to see some instant products."
```
Comment (louis-bompart) [inline packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-instant-products/atomic-commerce-search-box-instant-products.tsx:235]:
```suggestion
        "There doesn't seem to be any query suggestions configured. Make sure to include either an atomic-commerce-search-box-query-suggestions or atomic-commerce-search-box-recent-queries in your search box in order to see some instant products."
```

**Review by louis-bompart — APPROVED** [review]
lgtm, nitpik

File: packages/atomic/src/pages/examples/commerce-website/search.html [hunk 2]
```diff
@@ -29,8 +29,13 @@ <h1>Search page</h1>
         <atomic-commerce-interface type="search" style="max-width: 1000px; margin: auto">
           <atomic-search-layout>
             <atomic-layout-section section="search">
-              <atomic-commerce-search-box textarea></atomic-commerce-search-box>
-              <!-- <atomic-commerce-search-box redirection-url="https://docs.coveo.com/en/1458" textarea></atomic-commerce-search-box> -->
+              <atomic-commerce-search-box textarea>
```
Comment (olamothe) [inline packages/atomic/src/pages/examples/commerce-website/search.html:32]: Unrelated to the PR: `textarea` property is not doing anything/atomic commerce search box is already a text area by default.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/pages/examples/commerce-website/search.html (hunk 2 above)
Comment (louis-bompart) [reply packages/atomic/src/pages/examples/commerce-website/search.html:32]:
```suggestion
              <atomic-commerce-search-box>
```
quickwin

**Review by Spuffynism — APPROVED** [review]

**Pull request merged**